index.js

```diff
@@ -1,262 +1,515 @@
- * An ampersand.js model to represent a connection to a MongoDB database.
- * It does not actually talk to MongoDB.  It is just a higher-level
- * abstraction that prepares arguments for `MongoClient.connect`.
-**/
-var Connection = AmpersandModel.extend({
-  namespace: 'Connection',
-  idAttribute: 'instance_id',
-  props: {
-    /**
-     * User specified name for this connection.
-     */
-    name: {
-      type: 'string',
-      default: 'Local'
-    },
-    hostname: {
-      type: 'string',
-      default: 'localhost'
-    },
-    port: {
-      type: 'number',
-      default: 27017
-    },
-    mongodb_username: {
-      type: 'string',
-      default: null
-    },
-    mongodb_password: {
-      type: 'string',
-      default: null
-    },
-    /**
-     * Used when the user for authentication is stored
-      in another database using indirect authentication.
-     */
-    auth_source: {
-      type: 'string',
-      default: 'admin'
-    },
-    /**
-     * ## `null`
-     * No authentication.
-     #
-     * ## GSSAPI
-     * Kerberos
-     *
-     * ## PLAIN
-     * LDAP
-     *
-     * ## DEFAULT
-     * driver will autoselect SCRAM-SHA-1 | MONGODB-CR
-     * depending on the servers capabilities.
-     * @see https://github.com/mongodb/node-mongodb-native/pull/1299
-     */
-    auth_mechanism: {
-      type: 'string',
-      values: [
-        null,
-        'DEFAULT', 'SCRAM-SHA-1', 'MONGODB-CR',
-        'MONGODB-X509',
-        'GSSAPI',
-        'PLAIN'
-      ],
-      default: null
-    },
-    gssapi_service_name: {
-      type: 'string',
-      default: null
-    },
-    /**
-     * a.k.a `authdb`
-     */
-    database_name: {
-      type: 'string',
-      default: 'admin'
-    },
-    ssl: {
-      type: 'boolean',
-      default: false
-    },
-    /**
-     * Validate mongod server certificate against ca
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_validate: {
-      type: 'boolean',
-      default: false
-    },
-
-    /**
-     * Array of valid certificates either as Buffers or Strings
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_ca: {
-      type: 'array',
-      default: null
-    },
-
-    /**
-     * String or buffer containing the certificate we wish to present
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_cert: {
-      type: 'string',
-      default: null
-    },
-
-    /**
-     * String or buffer containing the certificate private key we wish to present
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_private_key: {
-      type: 'string',
-      default: null
-    },
-
-    /**
-     * String or buffer containing the certificate password
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_private_key_password: {
-      type: 'string',
-      default: null
+ * # Top-Level
+ */
+assign(props, {
```

Comment:
On a similar note, is there a reason we're using this rather than just setting the values from the start? I see how it lets you order things nicely, but it also seems to add unnecessary complexity.